Add option to embed checklist

sh-csg opened the pull request:
This adds an option to embed the checklist directly in the block

sh-csg committed:
Add option to embed checklist

davosmith commented:
Thanks for this @sh-csg - it looks great.

One thing I'd want to see, before merging the change, would be a simple Behat test that confirms it is working as expected and that it continues to work during future upgrades (I'm aware that the old code here does not have any existing Behat test coverage, but I'm wanting to improve on that as time goes on).

Something simple that generates a checklist with a few items, adds the block to a course, ticks off a couple of items in the checklist, then refreshes the page and confirms that it stays updated. Most of that can be easily cribbed from the existing Behat tests for mod_checklist.

Would it be possible for you to add such a test?

sh-csg commented:
Hi @davosmith ,
thanks a lot for your feedback! I added a small behat test and changed the GHA to install mod_checklist (for being able to use the steps defined there).
